src/__experimental__/components/checkbox/checkbox.component.js

```diff
@@ -1,31 +1,45 @@
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
 import tagComponent from '../../../utils/helpers/tags';
 import CheckboxStyle from './checkbox.style';
 import CheckableInput from '../checkable-input/checkable-input.component';
 import CheckboxSvg from './checkbox-svg.component';
-import withValidations from '../../../components/validations/with-validation.hoc';
+import withValidation from '../../../components/validations/with-validation.hoc';
 
 const Checkbox = ({
-  id, label, onChange, value, ...props
+  id, name, label, value, onChange, defaultChecked, checked, ...props
 }) => {
+  const checkedValue = checked === undefined ? false : checked;
+  const checkedStatus = (checked === undefined && defaultChecked !== undefined) ? defaultChecked : checkedValue;
+
```

Comment:
This unfortunately doesn't work 100% as a controlled component, because it does not react to the `checked` prop changes. You can check it out by having two checkboxes linked to one state field. Clicking on one, should also automatically check the other one, but it does not happen.

Reply:
Many thanks for spotting that bug. Yes you're correct, `setIsChecked()` is currently only called by `handleOnChange()` (when the user toggles the checkbox). But `setIsChecked()` also needs to be called when the `checked` prop changes. Fixed in 3bcbdc2 :thumbsup:

Reply:
No worries :)